### PR TITLE
Permissions page redux

### PR DESCRIPTION
**Pull request by lauriii**

### CODE REVIEW

File: src/components/05_pages/Permissions/Permissions.js [hunk 1]
```diff
@@ -26,67 +28,48 @@ export const filterPermissions = (input, permissions) =>
...
+    this.props.cancelTask();
   }
   onPermissionCheck = (roleName, permission) => {
     this.setState(prevState => ({
-      changedRoles: [...new Set(prevState.changedRoles).add(roleName).values()],
+      ...prevState,
+      changedRoles: Array.from(
+        new Set(prevState.changedRoles).add(roleName).values(),
+      ),
       roles: this.togglePermission(permission, roleName, prevState.roles),
```
Comment (mattgrill) [inline src/components/05_pages/Permissions/Permissions.js]: `prevState.roles` does not exist until you make the first change to permissions.

Comment (dawehner) [reply src/components/05_pages/Permissions/Permissions.js]: good catch!

**Comment by mattgrill** [comment]
👏

File: src/components/05_pages/Permissions/Permissions.js [hunk 2]
```diff
@@ -26,67 +28,52 @@ export const filterPermissions = (input, permissions) =>
...
   }
   componentWillUnmount() {
-    this.cancelFetch();
+    this.props.cancelTask();
   }
   onPermissionCheck = (roleName, permission) => {
     this.setState(prevState => ({
-      changedRoles: [...new Set(prevState.changedRoles).add(roleName).values()],
-      roles: this.togglePermission(permission, roleName, prevState.roles),
+      ...prevState,
```
[the remark this answers is not in the document]
Comment (lauriii) [reply src/components/05_pages/Permissions/Permissions.js]: Good catch! 👍

**Comment by dawehner** [comment]
We could close this too, I am not sure this is really that important.

**Comment by dawehner** [comment]
Let's close this PR for now :)

**Pull request closed without merging**